src/qcodes/dataset/descriptions/dependencies.py

```diff
@@ -63,163 +53,153 @@ def __init__(
         inferences: ParamSpecTree | None = None,
         standalones: tuple[ParamSpecBase, ...] = (),
     ):
-        dependencies = dependencies or {}
-        inferences = inferences or {}
-
-        deps_error = self.validate_paramspectree(dependencies)
-        if deps_error is not None:
-            try:
-                raise deps_error[0](deps_error[1])
-            except Exception as old_error:
-                raise ValueError("Invalid dependencies") from old_error
-
-        inffs_error = self.validate_paramspectree(inferences)
-        if inffs_error is not None:
-            try:
-                raise inffs_error[0](inffs_error[1])
-            except Exception as old_error:
-                raise ValueError("Invalid inferences") from old_error
-
-        link_error = self._validate_double_links(dependencies, inferences)
-        if link_error is not None:
-            error, mssg = link_error
-            raise error(mssg)
-
+        self._graph: nx.DiGraph[str] = nx.DiGraph()
+        self.add_dependencies(dependencies)
+        self.add_inferences(inferences)
+        self.add_standalones(standalones)
+
+    def add_paramspecs(self, paramspecs: list[ParamSpecBase]) -> None:
```

Comment:
```suggestion
    def add_paramspecs(self, paramspecs: Sequence[ParamSpecBase]) -> None:
```
Since this doesn't modify the input the type should be immutable